src/main/java/ch/njol/skript/expressions/ExprHoverList.java

```diff
@@ -0,0 +1,160 @@
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Hover List")
+@Description({"The list when you hover on the player counts of the server in the server list.",
+		"This can be changed using texts or players in a <a href='events.html#server_list_ping'>server list ping</a> event only. " +
+		"Adding players to the list means adding name of the players.",
+		"And note that, for example if there are 5 online players (includes <a href='#ExprOnlinePlayersCount'>fake online count</a>) " +
+		"in the server and the hover list is set to 3 values, Minecraft will show \"... and 2 more ...\" at end of the list.",
+		"",
+		"Requires PaperSpigot 1.12.2+."})
+@Examples({"on server list ping:",
+		"	clear the hover list",
+		"	add \"<light green>Welcome to the <orange>Minecraft <light green>server!\" to the hover list",
+		"	add \"\" to the hover list # A blank line",
+		"	add \"<light red>There are <orange>%online players count% <light red>online players!\""})
+@Since("INSERT VERSION")
+public class ExprHoverList extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprHoverList.class, String.class, ExpressionType.SIMPLE,
+				"[the] [custom] [(player|server)] (hover|sample) ([message] list|message)",
+				"[the] [custom] player [(hover|sample)] list");
+	}
+
+	@SuppressWarnings("null")
+	private Kleenean delay;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		boolean isServerPingEvent = ScriptLoader.isCurrentEvent(ServerListPingEvent.class);
+		boolean isPaperEvent = Skript.classExists("com.destroystokyo.paper.event.server.PaperServerListPingEvent") && ScriptLoader.isCurrentEvent(PaperServerListPingEvent.class);
+		if (isServerPingEvent) {
+			Skript.error("The hover list expression requires PaperSpigot 1.12.2+");
+			return false;
+		} else if (!isPaperEvent) {
+			Skript.error("The hover list expression can't be used outside of a server list ping event" + (isServerPingEvent ? " and requires PaperSpigot 1.12.2+" : ""));
+			return false;
+		}
+		delay = isDelayed;
+		return true;
+	}
+
+	@Override
+	public boolean isSingle() {
+		return false;
+	}
+
+	@Override
+	@Nullable
+	public String[] get(Event e) {
+		return ((PaperServerListPingEvent) e).getPlayerSample().stream()
+				.map(PlayerProfile::getName)
+				.toArray(String[]::new);
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (delay == Kleenean.TRUE) {
+			Skript.error("Can't change the hover list anymore after the server list ping event has already passed");
+			return null;
+		}
+		if (mode == Changer.ChangeMode.SET ||
+				mode == Changer.ChangeMode.ADD ||
+				mode == Changer.ChangeMode.REMOVE ||
+				mode == Changer.ChangeMode.DELETE ||
+				mode == Changer.ChangeMode.RESET)
+			return CollectionUtils.array(String[].class, Player[].class);
```

Comment:
Could you please consider using a switch statement here? I think it's rather a standard in the `acceptChange` method. Same thing in the other classes

Reply:
No only `if (mode != ChangeMode.REMOVE_ALL)` would be needed.

Reply:
Damn you and @Blueyescat don't really like placing commas in important places, do you *lol*

Anyway, probably a good catch indeed, *if* Njol used to write it this way too (or did he always use switches here?)

Reply:
If you care about possible new changers, you should use a switch; if you don't, you should use @TheLimeGlass 's method

Reply:
i will use switch for 3+ changer